app/src/forms/teammanagement/controller.js

```diff
@@ -1,13 +1,17 @@
+const EventEmitter = require('events');
+
 const Service = require('./service');
 
-class Controller {
+class Controller extends EventEmitter {
   constructor(resourceAccess) {
+    super();
     this._service = new Service(resourceAccess);
   }
 
   async processAccessRequest(req, res, next) {
     try {
       const response = await this._service.processAccessRequest(req.body, req.currentUser);
+      this.emit('onAccessRequest', response);
```

Comment:
What's the emitter pattern about? More useful then just calling service layer/whatever functions?

Reply:
I think this is leveraging node's event handler pattern IIRC. It's one way to insert extra functionality.

Reply:
Yeah just curious since the line above is also an async (albeit blocked with an await) call out to a function, and then this is a different way of calling out to do some work. So 2 different patterns next to each other.

Reply:
node EventEmitter, just easier than passing callbacks down.  I don't think we care if that is actioned or not, so no need to await.  Maybe not every form of TeamManagement will care and hook into that event.